### ISSUE TITLE
--incompatible fails with transient-switches

### ISSUE THREAD

**Issue by stephenwithav** [magit/transient#258]
I want `--defaults` to be incompatible with `--standalone`, `--routing`, AND `--style`, but the latter fails.

Choosing `--defaults` when `--standalone` or `--routing` is selected cancels `--defaults`, but none of (`--style`, `--style=`, `--style=%s`, or `--style=css`) cancels `--defaults`.

```elisp
(transient-define-suffix aangit-menu--ng-new (&optional args)
  :description "suffix"
  (interactive (list (transient-args transient-current-command)))
  ;; ng prompts interactively for unspecified prompts, which causes failures.
  ;; (shell-command "ng new %s %s" (car (last (string-split (car (dired-read-dir-and-switches "")) "/"))) (string-join args " "))
  (message "ng new %s %s" (car (last (string-split (car (dired-read-dir-and-switches "")) "/"))) (string-join args " "))
  )

(transient-define-argument aangit-menu--new-project-style ()
  :description "Style"
  :class transient-switches
  :key "-y"
  :argument-format "--style=%s"
  :argument-regexp ".+"
  :choices '("css" "scss" "sass" "less")
  )

(transient-define-prefix aangit-menu--new-project ()
  :incompatible '(
                  ("--defaults" "--standalone")
                  ("--defaults" "--routing")
                  ("--defaults" "--style=%s")
                  )
  :value '("--defaults")
  ["Switches"
   ("-d" "Defaults" "--defaults" :class transient-switch)
   ("-s" "Standalone" "--standalone" :class transient-switch)
   ("-r" "Routing" "--routing" :class transient-switch)
   (aangit-menu--new-project-style)
   ]
  ["Commands"
   ("n" "new" aangit-menu--ng-new)]
  )

(transient-define-prefix aangit-menu ()
  [["ng"
    ("n" "new" aangit-menu--new-project)
    ]])
```
version: Magit 97a95f7, Git 2.25.1, Emacs 29.1, gnu/linux

**Comment by stephenwithav** [comment]
This works, but it's a less ideal UX.

```elisp
(transient-define-argument aangit-menu--new-project-style ()
  :description "Style"
  :class transient-option
  :key "-y"
  :argument "--style="
  :choices '("css" "scss" "sass" "less")
  )

(transient-define-prefix aangit-menu--new-project ()
  :incompatible '(
                  ("--defaults" "--standalone")
                  ("--defaults" "--routing")
                  ("--defaults" "--style=")
                  )
  :value '("--defaults")
  ["Switches"
   ("-d" "Defaults" "--defaults" :class transient-switch)
   ("-s" "Standalone" "--standalone" :class transient-switch)
   ("-r" "Routing" "--routing" :class transient-switch)
   (aangit-menu--new-project-style)
   ]
  ["Commands"
   ("n" "new" aangit-menu--ng-new)]
  )
```

**Comment by tarsius** [comment]
Indeed, the `transient-infix-set :around transient-argument` method needs some work. It will probably a week or two until I get around to this, but you can continue doing it like this and assume the automatic disabling of incompatible arguments will start working in a not so far away future.

Except you should use this:

```elisp
  :argument-format "--style=%s"
  :argument-regexp "--style=\\(css\\|scss\\|sass\\|less\\)"
  :choices '("css" "scss" "sass" "less")
```

(Doing that will actually partially fix the issue. But I should rethink all of this, possibly making it unnecessary to specify `argument-regexp` when the other two slots are specified.)

**Comment by stephenwithav** [comment]
Thank you for working on this, @tarsius.

That `:argument-regexp` swap fails for me.  `--defaults` and `--style` can both be active.

**Comment by tarsius** [comment]
If implemented this now in ed5bd6fdc38dc586793315dff1be5c1643c96b4c. 2eb214a6ecbd632760706de9056441fb581ec5fa adds support for `--option=with-value` and updates the documentation.

Sorry this took a little longer than I had planned.
